versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -1146,21 +1146,27 @@ protected Hash tryMoveNamedReference(
         .build();
   }
 
-  /**
-   * Get database-specific 'strings' like column definitions for 'BLOB' column types.
-   *
-   * <ul>
-   *   <li>Index #0: column-type string for a 'BLOB' column.
-   *   <li>Index #1: column-type string for the string representation of a {@link Hash}
-   *   <li>Index #2: column-type string for key-prefix
-   *   <li>Index #3: column-type string for the string representation of a {@link Key}
-   *   <li>Index #4: column-type string for the string representation of a {@link NamedRef}
-   *   <li>Index #5: column-type string for the named-reference-type (single char)
-   *   <li>Index #6: column-type string for the contents-id
-   *   <li>Index #7: column-type string for an integer
-   * </ul>
-   */
-  protected abstract List<String> databaseSqlFormatParameters();
+  /** Get database-specific 'strings' like column definitions for 'BLOB' column types. */
+  protected abstract Map<SqlDataType, String> databaseSqlFormatParameters();
+
+  protected enum SqlDataType {
```

Comment:
Isn't this a "nessie" data type? WDYT about renaming to `NessieDataType` or perhaps `AdapterDataType`?

Reply:
renamed